### PR TITLE
added username to login functionality

### PR DESCRIPTION
**Pull request by RanSolo**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-  # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :exception
-
   before_action :configure_permitted_parameters, if: :devise_controller?
 
   private
 
   def configure_permitted_parameters
-    devise_parameter_sanitizer.for(:sign_up) << :username
+    devise_parameter_sanitizer.for(:sign_up) { |u| u.permit(:username, :email, :password, :password_confirmation, :remember_me) }
```
Comment (elizamarcum) [inline app/controllers/application_controller.rb:10]: Why did you switch this from using the push operator?

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<h2>Sign in</h2>
+
+<%= simple_form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
```
Comment (elizamarcum) [inline app/views/devise/sessions/new.html.erb]: Please convert this to haml.

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  <div class="form-inputs">
+    <%= f.input :email || :username, required: false, autofocus: true, label: 'Email/Username', name: 'Email/Username' %>
```
Comment (elizamarcum) [inline app/views/devise/sessions/new.html.erb]: the `:email || :username` section here means that the field will be for `:email`, since `||` returns the value of the first truthy argument.

**Comment by elizamarcum** [comment]
This looks like it's really close!

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,13 +1,9 @@
 class ApplicationController < ActionController::Base
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :exception
```
Comment (elizamarcum) [inline app/controllers/application_controller.rb:4]: Why was this line removed?

Comment (RanSolo) [inline app/controllers/application_controller.rb:4]: good old fashioned accident.

got it back in there.
